src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -1186,6 +1187,7 @@ public static partial class JsonMetadataServices
         public static System.Text.Json.Serialization.JsonConverter<T> GetEnumConverter<T>(System.Text.Json.JsonSerializerOptions options) where T : struct { throw null; }
         public static System.Text.Json.Serialization.JsonConverter<T?> GetNullableConverter<T>(System.Text.Json.JsonSerializerOptions options) where T : struct { throw null; }
         public static System.Text.Json.Serialization.JsonConverter<T?> GetNullableConverter<T>(System.Text.Json.Serialization.Metadata.JsonTypeInfo<T> underlyingTypeInfo) where T : struct { throw null; }
+        public static System.Text.Json.Serialization.JsonConverter<T> GetStringEnumConverter<T>(System.Text.Json.Serialization.JsonStringEnumConverter converterFactory, System.Text.Json.JsonSerializerOptions options) where T : struct { throw null; }
```

Comment:
Why doesn't this have `where T : struct, Enum`? That is what is in the [approved API](https://github.com/dotnet/runtime/issues/79311#issuecomment-1522146455).

Reply:
It would appear that the underlying converter doesn't have a `: Enum` constraint. Given that this matches the existing GetEnumConverter method, we should probably just update the proposal.

Reply:
@eiriktsarpalis the code for APICompat which is used to compare the contract vs the implementation is located in the sdk repo: https://github.com/dotnet/sdk/tree/main/src/ApiCompat

Please file an issue so that we can track this bug / feature request (unsure yet what it is).

> @carlossanlop @ViktorHofer is it safe to update the ref file to match the implementation?

Yes, that should be fine.

Reply:
Dumb question, but did you make sure to autogenerate the ref file @eiriktsarpalis ? https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/docs/coding-guidelines/updating-ref-source.md#L16